### PR TITLE
Modernize Biome Registration

### PR DESCRIPTION
**Pull request by AtomicBlom**
This is a request for comments.
At the time of writing, the Twilight Forest generates as nothing but ocean biomes because the biome registration is not yet done using the biome forge registry.

This PR represents the normal way I work with forge registries, using the new registry events and ObjectHolder annotations.

Some people prefer to use less verbose approaches, hence the RFC.

One thing to note is that the blocks and items need to also be updated because this event fires before preInit. I have a branch that is most of the way through updating items and blocks before I was forced to call it a night.

### CODE REVIEW

**Comment by williewillus** [comment]
This actually looks fine, two things:

1. When registering no need to put comments with the biome name above each one - the registry name makes it clear already

2. I'd name the holder `TFBiomes` like vanilla, `BiomeLibrary` sounds a bit weird

File: src/main/java/twilightforest/RegistryBiomeEvent.java [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+
+		private final IForgeRegistry<Biome> registry;
+
+		BiomeRegistry(IForgeRegistry<Biome> registry)
+		{
+
+			this.registry = registry;
+		}
+
+		public void register(String registryName, Biome biome, Type... biomeTypes)
```
Comment (liach) [inline src/main/java/twilightforest/RegistryBiomeEvent.java]: No need to be public.

File: src/main/java/twilightforest/library/BiomeLibrary.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+		clearing = null;
+		oakSavanna = null;
+		fireflyForest = null;
+		deepMushrooms = null;
+		darkForest = null;
+		enchantedForest = null;
+		fireSwamp = null;
+		darkForestCenter = null;
+		highlandsCenter = null;
+		thornlands = null;
```
Comment (liach) [inline src/main/java/twilightforest/library/BiomeLibrary.java]: Or simply create a fake biome `Biome fake = new Biome() {/* implement methods*/}` and set all fields to that fake one.

Comment (AtomicBlom) [reply src/main/java/twilightforest/library/BiomeLibrary.java]: Object holder won't set fields that aren't null iirc.

**Comment by AtomicBlom** [comment]
Good point on the comments, originally I was renaming the fields in the object holder, the comments helped me keep track of them, so they're not needed anymore

**Comment by williewillus** [comment]
Did you register the item forms of the blocks?

**Comment by AtomicBlom** [comment]
Yep, It's been moved into item registration.

**Comment by AtomicBlom** [comment]
With this PR, you can now load into the Twilight Forest (as opposed to the Twilight Ocean)

**Comment by williewillus** [comment]
If it works I'd say go ahead and merge it, looks fine to me

**Pull request merged**